core/templates/dev/head/domain/question/QuestionObjectFactory.js

```diff
@@ -97,6 +99,19 @@ oppia.factory('QuestionObjectFactory', [
       );
     };
 
+    Question.prototype.toBackendDict = function(newQuestion) {
```

Comment:
Why pass in newQuestion?

Reply:
It's just a flag so as to not add an id or version field when the question is a new one, since those are added in the backend controller.
Should I change name to isNewQuestion or similar?

Reply:
Yes, I think so. Also use `id: null` instead; make sure the list of keys is fixed and is the same in all cases.

Reply:
Actually, the backend controller throws an error if an ID field is even present. Should I change to check if it's None when creating a question?

Reply:
Yep, that sounds good!

Reply:
Done.